ocaml/xapi/dbsync_master.ml

```diff
@@ -149,8 +155,10 @@ let create_tools_sr __context name_label name_description sr_introduce maybe_cre
           ) (Db.SR.get_all ~__context)
       in
       match old_srs with
-      | sr :: _ ->
-        Db.SR.set_is_tools_sr ~__context ~self:sr ~value:true; sr
+      | sr :: others ->
+        Db.SR.set_is_tools_sr ~__context ~self:sr ~value:true;
+        List.iter destroy others; (* destroy bogus Tool SRs CA-300103 *)
```

Comment:
We want to do this cleanup of the "old" and "bogus" Tools SRs also if we did find a Tools SR (with `is_tools_sr = true`), so in the `| sr :: []` case.

By the way, the commit message is not quite right: this is not executed on pool join, but when the master xapi starts. Pool join is fixed by the previous commit already. This code is meant to fix things up for pools that already have the bogus tools SRs due the bug that the previous commit fixes.